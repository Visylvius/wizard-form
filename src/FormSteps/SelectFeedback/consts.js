const optionsArr = [
  { value: 'website', displayValue: 'Our Website' },
  { value: 'products', displayValue: 'Our Products' },
  { value: 'retail', displayValue: 'Our Retail Locations' },
  { value: 'service', displayValue: 'Our Customer Service' },
  { value: 'general', displayValue: 'Cole Haan, Generally' },
];

const SELECT_FEEDBACK = 'selectFeedback';

export { SELECT_FEEDBACK, optionsArr };
