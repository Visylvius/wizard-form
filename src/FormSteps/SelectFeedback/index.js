import React from 'react';

import Select from '../../Components/Select';
import Button from '../../Components/Button';
import { DEFAULT_VALUE } from '../../globalConsts';

import { optionsArr, SELECT_FEEDBACK } from './consts';
import './styles.scss';

const SelectFeedback = ({
  onChange,
  onButtonClick,
  selectedVal,
  currentStep,
  testId,
}) => {
  if (currentStep !== 2) {
    return null;
  }

  return (
    <>
      <Select
        selectTestId={testId}
        name={SELECT_FEEDBACK}
        optionsArr={optionsArr}
        selectedVal={selectedVal}
        onChange={onChange}
        label='Please choose a category you would like to leave feedback about.'
      />
      <Button
        data-testid='SelectFeedback-Button'
        buttonClasses='SelectFeedback-Button'
        disabled={selectedVal === DEFAULT_VALUE}
        onClick={onButtonClick}
        shouldCenter
      >
        Next
      </Button>
    </>
  );
};

export default SelectFeedback;
