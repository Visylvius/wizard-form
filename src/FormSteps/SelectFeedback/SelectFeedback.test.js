import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SelectFeedback from './index';

describe('SelectFeedback (Unit)', () => {
  const mockOnChange = jest.fn();
  const mockOnClick = jest.fn();
  const renderSelectFeedback = props =>
    render(
      <SelectFeedback
        testId='SelectFeedback-select'
        currentStep={2}
        onChange={mockOnChange}
        onButtonClick={mockOnClick}
        selectedVal='default'
        {...props}
      />,
    );

  describe('render', () => {
    test('should render properly', () => {
      const tree = renderer.create(
        <SelectFeedback
          currentStep={2}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });

    test('should render null if the step is not 5', () => {
      const tree = renderer.create(
        <SelectFeedback
          currentStep={3}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Intergration', () => {
    test('should properly call onChange', async () => {
      const { getByTestId, getByText } = renderSelectFeedback();

      fireEvent.change(getByTestId('SelectFeedback-select'), {
        target: { value: 'products' },
      });

      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
      expect(getByText('Our Products')).toBeInTheDocument();
    });

    test('should properly call onClick', async () => {
      const { getByTestId } = renderSelectFeedback({ selectedVal: 'service' });

      userEvent.click(getByTestId('SelectFeedback-Button'));

      await waitFor(() => expect(mockOnClick).toHaveBeenCalled());
    });

    test('should not call onClick when the button is disabled', async () => {
      const { getByTestId } = renderSelectFeedback();

      userEvent.click(getByTestId('SelectFeedback-Button'));

      await waitFor(() => expect(mockOnClick).not.toHaveBeenCalled());
    });
  });
});
