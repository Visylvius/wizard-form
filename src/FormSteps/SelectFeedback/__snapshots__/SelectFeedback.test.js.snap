// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFeedback (Unit) render should render null if the step is not 5 1`] = `null`;

exports[`SelectFeedback (Unit) render should render properly 1`] = `
Array [
  <label
    className="Select-label"
  >
    Please choose a category you would like to leave feedback about.
  </label>,
  <select
    className="Select-mainWrapper"
    name="selectFeedback"
    onChange={[MockFunction]}
    value="default"
  >
    <option
      defaultValue={true}
      disabled={true}
      value="default"
    >
      Choose...
    </option>
    <option
      data-testid="SelectOption-0"
      value="website"
    >
      Our Website
    </option>
    <option
      data-testid="SelectOption-1"
      value="products"
    >
      Our Products
    </option>
    <option
      data-testid="SelectOption-2"
      value="retail"
    >
      Our Retail Locations
    </option>
    <option
      data-testid="SelectOption-3"
      value="service"
    >
      Our Customer Service
    </option>
    <option
      data-testid="SelectOption-4"
      value="general"
    >
      Cole Haan, Generally
    </option>
  </select>,
  <div
    className="Button-container"
  >
    <button
      className="Button-component SelectFeedback-Button"
      data-testid="SelectFeedback-Button"
      disabled={true}
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>,
]
`;
